### PR TITLE
Feat/manager 12295 Tests E2E

### PR DESCRIPTION
**Pull request by Kranysys**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/EPIC-MANAGER-11852`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | feat/MANAGER-12295
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

## Description

E2E tests for the revamp of the catalog

## Related

Manager-11853 Manager-11857 Manager-12293 Manager-12298 Manager-11852

### CODE REVIEW

File: packages/manager/apps/catalog-revamp/src/e2e/FilterAndSearchBehavior.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  const checkboxeAIMachineLearning = checkboxes[6];
+  if (checkboxeAIMachineLearning) await checkboxeAIMachineLearning.click();
+
+  const applyFiltersButton = buttons[1];
+  if (applyFiltersButton) await applyFiltersButton.click();
+
+  await page.waitForTimeout(1000);
+
+  const products = await page.$$('msc-tile');
+  expect(products.length).toBe(0); // 0 products : no result
```
Comment (aboungnaseng-ovhcloud) [inline packages/manager/apps/catalog-revamp/src/e2e/FilterAndSearchBehavior.test.ts:74]: Can you test if the message is displayed pls

Comment (Kranysys) [reply packages/manager/apps/catalog-revamp/src/e2e/FilterAndSearchBehavior.test.ts]: Done

File: packages/manager/apps/catalog-revamp/e2e/FilterAndSearch.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    expect(await product.$('.tile-title')).toBeTruthy();
+  });
+
+  await Promise.all(checks);
+};
+
+test('should filter results based on Bare Metal Cloud universe', async ({
+  page,
+}) => {
+  await clickButtonByIndex(page, 0);
```
Comment (aboungnaseng-ovhcloud) [inline packages/manager/apps/catalog-revamp/e2e/FilterAndSearch.e2e.ts:52]: we can add 'name'  attribute on button or checkbox to be more specific for the test ?

Comment (Kranysys) [reply packages/manager/apps/catalog-revamp/e2e/FilterAndSearch.e2e.ts]: Done

File: packages/manager/apps/catalog-revamp/tsconfig.json [hunk 3]
```diff
@@ -23,6 +23,6 @@
       "@playwright-helpers/*": ["../../../../playwright-helpers/*"]
     }
   },
-  "include": ["src"],
+  "include": ["src", "e2e"],
```
[the remark this answers is not in the document]
Comment (Kranysys) [reply packages/manager/apps/catalog-revamp/tsconfig.json]: Done, added tsconfig.e2e.json and pretest in package.json and the folder in gitignore

**Pull request merged**